libpromises/expand.c

```diff
@@ -647,6 +647,46 @@ void BundleResolve(EvalContext *ctx, const Bundle *bundle)
         "Resolving classes and variables in 'bundle %s %s'",
         bundle->type, bundle->name);
 
+    /* first check if some variables were injected remotely into this bundle and
+     * remove them (CFE-1915) */
+    const Seq *remote_var_promises = EvalContextGetRemoteVarPromises(ctx);
```

Comment:
This needs to be limited to only promises that might be injecting variables into this bundle not all the other bundles.

Reply:
Off the top of my head: commands (via modules), classes type promises (some define classic arrays as a side effect), vars, meta, defaults